sources/languages/euphoria/description.md

```diff
@@ -38,11 +38,58 @@ is actually this:
 {65, 66, 67}
 ```
 
-Also according to [Wikipedia](https://en.wikipedia.org/wiki/Euphoria_(programming_language)#Parameter_passing)):
+[User-specified data types](https://openeuphoria.org/docs/lang_decl.html#_123_userdefinedtypes)
+can be done based on these four data types. The way this is done is to declare a type
+function that validates the input value. For example, if you wanted to define an
+`hour` type that only allowed integers from 0 to 23, you would do this:
+
+```euphoria
```

Comment:
Okay, this is really cool! Something like this would help me with some of the game development I've done in the past. Having custom bounded types are really nice and usually take more work in other languages.